src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -414,6 +461,12 @@ public static void register() throws Exception {
             Jenkins.get().setNoUsageStatistics(true);
             DownloadService.neverUpdate = true;
             UpdateSite.neverUpdate = true;
+            System.err.println("RealJenkinsRule ready");
+            Timer.get().schedule(JenkinsRule::dumpThreads, 2, TimeUnit.MINUTES);
+            Timer.get().schedule(() -> {
+                JenkinsRule.dumpThreads();
+                System.exit(1);
+            }, 5, TimeUnit.MINUTES);
```

Comment:
err can't this be configurable in some way, this test takes about 5 minutes to run.

https://github.com/jenkinsci/azure-container-agents-plugin/blob/dev/src/test/java/com/microsoft/jenkins/containeragents/AciCloudIT.java#L51